FieldOpt/Hdf5SummaryReader/hdf5_summary_reader.cpp

```diff
@@ -8,27 +8,176 @@ Hdf5SummaryReader::Hdf5SummaryReader(const std::string file_path)
   DATASET_NAME_TIMES("TIMES"),
   GROUP_NAME_FLOW_TRANSPORT("FLOW_TRANSPORT"),
   DATASET_NAME_ACTIVE_CELLS("ACTIVE_CELLS"),
-  DATASET_NAME_WELL_STATES("WELL_STATES")
+  DATASET_NAME_WELL_STATES("WELL_STATES"),
+  DATASET_NAME_PRESSURE("PTZ"),
+  DATASET_NAME_SATURATION("GRIDPROPTIME")
 {
     readTimeVector(file_path);
-    readActiveCells(file_path);
     readWellStates(file_path);
+    readActiveCells(file_path);
+    readReservoirPressure(file_path);
+    readSaturation(file_path);
 }
 
-void Hdf5SummaryReader::readActiveCells(std::string file_path) {
+void Hdf5SummaryReader::readSaturation(std::string file_path) {
+
+    // Check file exists
+    H5File file(file_path, H5F_ACC_RDONLY);
+    Group group = Group(file.openGroup(GROUP_NAME_FLOW_TRANSPORT));
+    auto dataset_exists = H5Lexists(group.getId(), "GRIDPROPTIME", H5F_ACC_RDONLY);
+    // std::cout << "dataset_exists = " << dataset_exists << std::endl;
+
+    std::vector<std::vector<double>> sgas, soil, swat;
+
+    /*!
+    \todo Overall, this needs a better implementation, e.g., read soil and sgas
+    at the same time from the H5 group, instead of one vector at a time, which
+    appears inefficient... on the other hand, this reading needs to be made 
+    flexible/robust against the different phase combinations that might exist
+    in GRIDPROPTIME, e.g., soil/sgas, soil/sgas/swat, soil/swat, etc. 
+    */
+
+    /*!
+    \todo Later: to save space, load saturation data as additional columns in 
+    PTZ (replacing current compositional columns? comtrolled by custom KEYWORD?),
+    and remove GRIDPROPTIME completely
+    */
+
+    if (dataset_exists) {
+
+        // std::cout << "number of phases = " << number_of_phases() << std::endl;
+
+        hsize_t SOIL, SWAT, SGAS;
+        if (number_of_phases() < 3){
+            SGAS = 0;
+            SOIL = 2; // col: 3
+            SWAT = 1; // col: 2
+        }else{
+            SGAS = 1; // col: 2
+            SOIL = 2; // col: 3
+            SWAT = 3; // col: 4
+        }
+
+        sgas_ = getSaturation(group, SGAS);
+        soil_ = getSaturation(group, SOIL);
+        swat_ = getSaturation(group, SWAT);
+
+    }else{
+
+        sgas_ = reservoir_pressure();
+        soil_ = reservoir_pressure();
+        swat_ = reservoir_pressure();
+    }
+}
+
+std::vector<std::vector<double>> Hdf5SummaryReader::getSaturation(Group group, hsize_t sat_type) {
+
+    // Read the file
+    DataSet dataset = DataSet(group.openDataSet(DATASET_NAME_SATURATION));
+    DataSpace dataspace = dataset.getSpace();
+    hsize_t dims[3];
+    dataspace.getSimpleExtentDims(dims, NULL);
+
+    std::vector<double> vector;
+    vector.resize(dims[0] * dims[2]);
+
+    // std::cout << "sat_type = " << (int)sat_type << std::endl;
```

Comment:
Remove debug prints or check some verbosity flag, don't just comment them out. If it's something that's very important, it should also be checked with a unit test.